compiler/rustc_ast/src/mut_visit.rs

```diff
@@ -1014,19 +1010,18 @@ pub fn noop_visit_crate<T: MutVisitor>(krate: &mut Crate, vis: &mut T) {
             id: DUMMY_NODE_ID,
             vis: item_vis,
             span,
-            kind: ItemKind::Mod(module),
+            kind: ItemKind::Mod(Mod { inner: span, unsafety: Unsafe::No, items, inline: true }),
```

Comment:
Constructing this synthetic `Mod` seems kind of unfortunate, but it looks like it's necessary for the `flat_map_item` handling below.

Reply:
I don't like this either and left a FIXME above.
I'll try to investigate removing this some time later, together with similar code in `fn expand_crate`.